src/main/java/BoxedTypesEqualsProcessor.java

```diff
@@ -0,0 +1,54 @@
+import spoon.processing.AbstractProcessor;
+import spoon.reflect.code.CtCodeSnippetExpression;
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.code.BinaryOperatorKind;
+import spoon.reflect.code.CtBinaryOperator;
+
+public class BoxedTypesEqualsProcessor extends AbstractProcessor<CtElement> {
+
+    public BoxedTypesEqualsProcessor(String projectKey) throws Exception {
+        ParseAPI.parse(4973,"",projectKey);
+    }
+
+    @Override
+
+    public boolean isToBeProcessed(CtElement candidate)
+    {
+        if(candidate==null)
+        {
+            return false;
+        }
+        if (candidate instanceof CtBinaryOperator){
+            CtBinaryOperator op = (CtBinaryOperator)candidate;
+            if(op.getKind() == BinaryOperatorKind.EQ || op.getKind() == BinaryOperatorKind.NE){
+                CtExpression left = op.getLeftHandOperand();
+                CtExpression right = op.getRightHandOperand();
+                /*
+                The reason we don't check for the case where one variable is boxed is because Java implicitly unboxes
+                the boxed type to primitive, making the == check fine. See JLS #5.6.2:
+                https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.6.2
+                */
+                String nullType = getFactory().Code().createCodeSnippetExpression("").getType().NULL_TYPE_NAME;
```

Comment:
```suggestion
                String nullType = getFactory().Type().NULL_TYPE.getQualifiedName();
```

Reply:
Agreed that's a nicer solution. Commit incoming.